src/nvim/api/ui.c

```diff
@@ -170,20 +173,51 @@ void nvim_ui_set_option(uint64_t channel_id, String name,
   }
 }
 
-static void ui_set_option(UI *ui, String name, Object value, Error *error) {
-  if (strcmp(name.data, "rgb") == 0) {
+static void ui_set_option(UI *ui, String name, Object value, Error *error)
+{
+  if (strequal(name.data, "rgb")) {
     if (value.type != kObjectTypeBoolean) {
       api_set_error(error, kErrorTypeValidation, "rgb must be a Boolean");
       return;
     }
     ui->rgb = value.data.boolean;
-  } else if (strcmp(name.data, "popupmenu_external") == 0) {
+  } else if (strequal(name.data, "popupmenu_external")) {
+    // LEGACY: Deprecated option, use `ui_ext` instead.
     if (value.type != kObjectTypeBoolean) {
       api_set_error(error, kErrorTypeValidation,
                     "popupmenu_external must be a Boolean");
       return;
     }
-    ui->pum_external = value.data.boolean;
+    ui->ui_ext[kUIPopupmenu] = value.data.boolean;
+  } else if (strequal(name.data, "ui_ext")) {
```

Comment:
Would you prefer a bunch of `ext_foo` fields? I'm fine with that, I suppose, the pesudo-namespacing seemed like a signal that they shouldn't be separate options. Also the documentation for each one will be very redundant.